Add a way for LocalOverrides.xcconfig to disable all VerifyModule/TAPI checks

charliewolfe opened the pull request:
#### c8c7c5e7598ba9dbe5f857852662cc2862a17974
<pre>
Add a way for LocalOverrides.xcconfig to disable all VerifyModule/TAPI checks
<a href="https://bugs.webkit.org/show_bug.cgi?id=285830">https://bugs.webkit.org/show_bug.cgi?id=285830</a>
<a href="https://rdar.apple.com/142800798">rdar://142800798</a>

Reviewed by Elliott Williams and Wenson Hsieh.

This lets LocalOverrides.xcconfig disable all remaining VerifyModule/TAPI checks, similar to the change
done for JSC in 284179@main.

This makes my local incremental build ~45 seconds faster.

* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/SwiftOverlay/Configurations/WebKitSwiftOverlay.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:

Canonical link: <a href="https://commits.webkit.org/288819@main">https://commits.webkit.org/288819@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3f4e3e8fcdcde96eb42996e79d3c3b539ac0e5b9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/4128 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38808 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89585 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35514 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86588 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4213 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/12113 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65717 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23561 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87549 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/3184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76773 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46011 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3132 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31004 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34562 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/74058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31779 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90966 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11770 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/8582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74169 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12006 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72597 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73370 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17720 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16155 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/3205 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13088 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11722 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/17198 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11571 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15047 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13344 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebGPU/Configurations/WebGPU.xcconfig:
```diff
@@ -66,7 +66,10 @@ GCC_PREPROCESSOR_DEFINITIONS_OS_UNFAIR_LOCK_INLINE_ENABLE_WEBGPU_SWIFT = OS_UNFA
 MODULEMAP_FILE = WebGPU/WebGPU.modulemap;
 
-SUPPORTS_TEXT_BASED_API = YES;
+SUPPORTS_TEXT_BASED_API = $(SUPPORTS_TEXT_BASED_API_($OVERRIDE_SUPPORTS_TEXT_BASED_API));
```
Was this intended to be `$(OVERRIDE_SUPPORTS_TEXT_BASED_API)`?

```suggestion
SUPPORTS_TEXT_BASED_API = $(SUPPORTS_TEXT_BASED_API_$(OVERRIDE_SUPPORTS_TEXT_BASED_API));
```

charliewolfe replied:
Whoops. Yes, will fix.

whsieh commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -32,7 +32,10 @@ PRODUCT_BUNDLE_IDENTIFIER = com.apple.$(PRODUCT_NAME:rfc1034identifier);
 
 DEFINES_MODULE = YES;
-ENABLE_MODULE_VERIFIER = $(USE_INTERNAL_SDK);
+ENABLE_MODULE_VERIFIER = $(ENABLE_MODULE_VERIFIER_($OVERRIDE_ENABLE_MODULE_VERIFIER));
```
Looks like this one is still `$OVERRIDE_ENABLE_MODULE_VERIFIER`
```suggestion
ENABLE_MODULE_VERIFIER = $(ENABLE_MODULE_VERIFIER_$(OVERRIDE_ENABLE_MODULE_VERIFIER));
```

emw-apple reviewed (approved):
(Copied from Slack discussion)

Philosophically, I think that building should only do work that advances toward a build product, so I don’t like our reliance on enabled-by-default verifiers.

But I also don’t like turbo buttons in the build – it’s a tax on people who don’t know about the “go fast” mode, and if everyone knows about it, we’re lying to ourselves about the level of verification we build with.

I think a good goal to work towards longer term is making these verifiers off by default for development, while keeping it easy for people to understand and reproduce failures.

Some ideas that might help:
- Make EWS able to recognize verifier failures separate from normal build failures, and have  the “build failed” message say how to reproduce the failure.
- Write a script that runs verifiers manually over a build product directory, and then run that as a “test” (and make it clear how to run it yourself)
- Do something like this^ but also run it as a pre-commit / pre-PR check. Would take some clever logic to avoid actually trying to build in those stages

charliewolfe committed:
Add a way for LocalOverrides.xcconfig to disable all VerifyModule/TAPI checks
https://bugs.webkit.org/show_bug.cgi?id=285830
rdar://142800798

Reviewed by Elliott Williams and Wenson Hsieh.

This lets LocalOverrides.xcconfig disable all remaining VerifyModule/TAPI checks, similar to the change
done for JSC in 284179@main.

This makes my local incremental build ~45 seconds faster.

* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebInspectorUI/Configurations/WebInspectorUIFramework.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/SwiftOverlay/Configurations/WebKitSwiftOverlay.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:

Canonical link: https://commits.webkit.org/288819@main
